from PyQt6 import QtCore, QtWidgets

class Ui_Form(object):
    def setupUi(self, Form: QtWidgets.QWidget):
        Form.setObjectName("Form")
        Form.setEnabled(True)
        Form.resize(1024, 900)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Policy.Preferred, QtWidgets.QSizePolicy.Policy.Maximum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(Form.sizePolicy().hasHeightForWidth())
        Form.setSizePolicy(sizePolicy)
        Form.setMinimumSize(QtCore.QSize(700, 665))
        self.verticalLayout_11 = QtWidgets.QVBoxLayout(Form)
        self.verticalLayout_11.setObjectName("verticalLayout_11")
        self.groupBox_4 = QtWidgets.QGroupBox(parent=Form)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Policy.Preferred, QtWidgets.QSizePolicy.Policy.Maximum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.groupBox_4.sizePolicy().hasHeightForWidth())
        self.groupBox_4.setSizePolicy(sizePolicy)
        self.groupBox_4.setObjectName("groupBox_4")
        self.horizontalLayout_5 = QtWidgets.QHBoxLayout(self.groupBox_4)
        self.horizontalLayout_5.setObjectName("horizontalLayout_5")
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.verticalLayout_4 = QtWidgets.QVBoxLayout()
        self.verticalLayout_4.setObjectName("verticalLayout_4")
        self.trf_file = QtWidgets.QLineEdit(parent=self.groupBox_4)
        self.trf_file.setReadOnly(True)
        self.trf_file.setObjectName("trf_file")
        self.verticalLayout_4.addWidget(self.trf_file)
        self.irf_file = QtWidgets.QLineEdit(parent=self.groupBox_4)
        self.irf_file.setReadOnly(True)
        self.irf_file.setObjectName("irf_file")
        self.verticalLayout_4.addWidget(self.irf_file)
        self.horizontalLayout_2.addLayout(self.verticalLayout_4)
        self.verticalLayout_7 = QtWidgets.QVBoxLayout()
        self.verticalLayout_7.setObjectName("verticalLayout_7")
        self.trf_browse = QtWidgets.QPushButton(parent=self.groupBox_4)
        self.trf_browse.setAcceptDrops(True)
        self.trf_browse.setObjectName("trf_browse")
        self.verticalLayout_7.addWidget(self.trf_browse)
        self.irf_browse = QtWidgets.QPushButton(parent=self.groupBox_4)
        self.irf_browse.setAcceptDrops(True)
        self.irf_browse.setObjectName("irf_browse")
        self.verticalLayout_7.addWidget(self.irf_browse)
        self.horizontalLayout_2.addLayout(self.verticalLayout_7)
        self.horizontalLayout_5.addLayout(self.horizontalLayout_2)
        self.verticalLayout_11.addWidget(self.groupBox_4)
        self.groupBox = QtWidgets.QGroupBox(parent=Form)
        self.groupBox.setObjectName("groupBox")
        self.horizontalLayout_4 = QtWidgets.QHBoxLayout(self.groupBox)
        self.horizontalLayout_4.setObjectName("horizontalLayout_4")
        self.verticalGroupBox = QtWidgets.QGroupBox(parent=self.groupBox)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Policy.Maximum, QtWidgets.QSizePolicy.Policy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.verticalGroupBox.sizePolicy().hasHeightForWidth())
        self.verticalGroupBox.setSizePolicy(sizePolicy)
        self.verticalGroupBox.setObjectName("verticalGroupBox")
        self.verticalLayout_5 = QtWidgets.QVBoxLayout(self.verticalGroupBox)
        self.verticalLayout_5.setSizeConstraint(QtWidgets.QLayout.SizeConstraint.SetDefaultConstraint)
        self.verticalLayout_5.setContentsMargins(-1, 10, -1, -1)
        self.verticalLayout_5.setObjectName("verticalLayout_5")
        self.horizontalLayout = QtWidgets.QHBoxLayout()
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.verticalLayout_2 = QtWidgets.QVBoxLayout()
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.label = QtWidgets.QLabel(parent=self.verticalGroupBox)
        self.label.setObjectName("label")
        self.verticalLayout_2.addWidget(self.label)
        self.label_2 = QtWidgets.QLabel(parent=self.verticalGroupBox)
        self.label_2.setObjectName("label_2")
        self.verticalLayout_2.addWidget(self.label_2)
        self.label_5 = QtWidgets.QLabel(parent=self.verticalGroupBox)
        self.label_5.setObjectName("label_5")
        self.verticalLayout_2.addWidget(self.label_5)
        self.label_6 = QtWidgets.QLabel(parent=self.verticalGroupBox)
        self.label_6.setObjectName("label_6")
        self.verticalLayout_2.addWidget(self.label_6)
        self.label_24 = QtWidgets.QLabel(parent=self.verticalGroupBox)
        self.label_24.setObjectName("label_24")
        self.verticalLayout_2.addWidget(self.label_24)
        self.horizontalLayout.addLayout(self.verticalLayout_2)
        self.verticalLayout = QtWidgets.QVBoxLayout()
        self.verticalLayout.setObjectName("verticalLayout")
        self.binSize_widg = QtWidgets.QDoubleSpinBox(parent=self.verticalGroupBox)
        self.binSize_widg.setMaximum(9999999.0)
        self.binSize_widg.setObjectName("binSize_widg")
        self.verticalLayout.addWidget(self.binSize_widg)
        self.maxIter_widg = QtWidgets.QSpinBox(parent=self.verticalGroupBox)
        self.maxIter_widg.setMinimum(1000)
        self.maxIter_widg.setMaximum(100000000)
        self.maxIter_widg.setSingleStep(1000)
        self.maxIter_widg.setProperty("value", 5000)
        self.maxIter_widg.setObjectName("maxIter_widg")
        self.verticalLayout.addWidget(self.maxIter_widg)
        self.startOffset_widg = QtWidgets.QSpinBox(parent=self.verticalGroupBox)
        self.startOffset_widg.setMaximum(999999)
        self.startOffset_widg.setObjectName("startOffset_widg")
        self.verticalLayout.addWidget(self.startOffset_widg)
        self.maxTime_widg = QtWidgets.QSpinBox(parent=self.verticalGroupBox)
        self.maxTime_widg.setMaximum(9999999)
        self.maxTime_widg.setProperty("value", 1000)
        self.maxTime_widg.setObjectName("maxTime_widg")
        self.verticalLayout.addWidget(self.maxTime_widg)
        self.expCount_widg = QtWidgets.QSpinBox(parent=self.verticalGroupBox)
        self.expCount_widg.setWrapping(False)
        self.expCount_widg.setMinimum(1)
        self.expCount_widg.setMaximum(4)
        self.expCount_widg.setProperty("value", 1)
        self.expCount_widg.setObjectName("expCount_widg")
        self.verticalLayout.addWidget(self.expCount_widg)
        self.horizontalLayout.addLayout(self.verticalLayout)
        self.verticalLayout_5.addLayout(self.horizontalLayout)
        self.label_3 = QtWidgets.QLabel(parent=self.verticalGroupBox)
        self.label_3.setObjectName("label_3")
        self.verticalLayout_5.addWidget(self.label_3)
        self.horizontalLayout_6 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_6.setObjectName("horizontalLayout_6")
        self.max_x = QtWidgets.QSlider(parent=self.verticalGroupBox)
        self.max_x.setMaximum(300)
        self.max_x.setProperty("value", 20)
        self.max_x.setSliderPosition(20)
        self.max_x.setOrientation(QtCore.Qt.Orientation.Horizontal)
        self.max_x.setInvertedAppearance(False)
        self.max_x.setInvertedControls(False)
        self.max_x.setTickPosition(QtWidgets.QSlider.TickPosition.TicksBelow)
        self.max_x.setTickInterval(100)
        self.max_x.setObjectName("max_x")
        self.horizontalLayout_6.addWidget(self.max_x)
        self.max_x_out = QtWidgets.QSpinBox(parent=self.verticalGroupBox)
        self.max_x_out.setButtonSymbols(QtWidgets.QAbstractSpinBox.ButtonSymbols.NoButtons)
        self.max_x_out.setMaximum(300)
        self.max_x_out.setProperty("value", 20)
        self.max_x_out.setObjectName("max_x_out")
        self.horizontalLayout_6.addWidget(self.max_x_out)
        self.verticalLayout_5.addLayout(self.horizontalLayout_6)
        self.logFit_widg = QtWidgets.QCheckBox(parent=self.verticalGroupBox)
        self.logFit_widg.setEnabled(True)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Policy.Fixed, QtWidgets.QSizePolicy.Policy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.logFit_widg.sizePolicy().hasHeightForWidth())
        self.logFit_widg.setSizePolicy(sizePolicy)
        self.logFit_widg.setCheckable(True)
        self.logFit_widg.setChecked(False)
        self.logFit_widg.setObjectName("logFit_widg")
        self.verticalLayout_5.addWidget(self.logFit_widg)
        self.plotIRF_widg = QtWidgets.QCheckBox(parent=self.verticalGroupBox)
        self.plotIRF_widg.setCheckable(True)
        self.plotIRF_widg.setChecked(False)
        self.plotIRF_widg.setObjectName("plotIRF_widg")
        self.verticalLayout_5.addWidget(self.plotIRF_widg)
        self.scaled_widg = QtWidgets.QCheckBox(parent=self.verticalGroupBox)
        self.scaled_widg.setCheckable(True)
        self.scaled_widg.setChecked(True)
        self.scaled_widg.setObjectName("scaled_widg")
        self.verticalLayout_5.addWidget(self.scaled_widg)
        self.fit_button = QtWidgets.QPushButton(parent=self.verticalGroupBox)
        self.fit_button.setObjectName("fit_button")
        self.verticalLayout_5.addWidget(self.fit_button)
        self.label_4 = QtWidgets.QLabel(parent=self.verticalGroupBox)
        self.label_4.setObjectName("label_4")
        self.verticalLayout_5.addWidget(self.label_4)
        self.text_output = QtWidgets.QTextEdit(parent=self.verticalGroupBox)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.text_output.sizePolicy().hasHeightForWidth())
        self.text_output.setSizePolicy(sizePolicy)
        self.text_output.setObjectName("text_output")
        self.verticalLayout_5.addWidget(self.text_output)
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.csv_browse = QtWidgets.QPushButton(parent=self.verticalGroupBox)
        self.csv_browse.setAcceptDrops(True)
        self.csv_browse.setObjectName("csv_browse")
        self.horizontalLayout_3.addWidget(self.csv_browse)
        self.verticalLayout_5.addLayout(self.horizontalLayout_3)
        self.horizontalLayout_4.addWidget(self.verticalGroupBox)
        self.frame = QtWidgets.QFrame(parent=self.groupBox)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Policy.Minimum, QtWidgets.QSizePolicy.Policy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.frame.sizePolicy().hasHeightForWidth())
        self.frame.setSizePolicy(sizePolicy)
        self.frame.setFrameShape(QtWidgets.QFrame.Shape.StyledPanel)
        self.frame.setFrameShadow(QtWidgets.QFrame.Shadow.Raised)
        self.frame.setObjectName("frame")
        self.horizontalLayout_4.addWidget(self.frame)
        self.verticalLayout_11.addWidget(self.groupBox)

        self.retranslateUi(Form)
        QtCore.QMetaObject.connectSlotsByName(Form)

    def retranslateUi(self, Form: QtWidgets.QWidget):
        _translate = QtCore.QCoreApplication.translate
        Form.setWindowTitle(_translate("Form", "Lifetime Fitting"))
        self.groupBox_4.setTitle(_translate("Form", "Input"))
        self.trf_browse.setText(_translate("Form", "Browse for TRF"))
        self.irf_browse.setText(_translate("Form", "Browse for IRF"))
        self.groupBox.setTitle(_translate("Form", "Fitting"))
        self.label.setText(_translate("Form", "Bin Width (ps)"))
        self.label_2.setText(_translate("Form", "Maximum Fitting Iterations"))
        self.label_5.setText(_translate("Form", "Offset for the Starting Bin"))
        self.label_6.setText(_translate("Form", "Maximum Time to Fit (ns)"))
        self.label_24.setText(_translate("Form", "Number of Decays to fit"))
        self.label_3.setText(_translate("Form", "Maximum Time to Plot (ns)"))
        self.logFit_widg.setText(_translate("Form", "Fit in Log Space"))
        self.plotIRF_widg.setText(_translate("Form", "Plot the IRF"))
        self.scaled_widg.setText(_translate("Form", "Plot the Decays scaled by Their Coefficients"))
        self.fit_button.setText(_translate("Form", "Fit"))
        self.label_4.setText(_translate("Form", "Output"))
        self.csv_browse.setText(_translate("Form", "Save CSV"))